test/unit/attributes.js

```diff
@@ -4,127 +4,306 @@ import helpers from '../lib/helpers';
+          }
+        }
+      });
+
+      new MyEl().getAttribute('test').should.equal('true');
+    });
+
+    it('should allow a callback to return a default value', function () {
+      var MyEl = skate('my-el', {
+        attributes: {
+          test: {
+            default: function () {
+              return 'true';
+            }
+          }
+        }
+      });
+
+      new MyEl().getAttribute('test').should.equal('true');
+    });
+
+    it('should not override if already specified', function () {
+      var MyEl = skate('my-el', {
+        attributes: {
+          test: {
+            default: 'true'
+          }
+        }
+      });
+
+      skate.init(helpers.fixture('<my-el test="false"></my-el>'))
+      .firstChild
+      .test
+      .should
+      .equal('false');
+    });
+  });
+
+  describe('should define properties for all watched attributes', function (done) {
+    var myEl;
     var created = false;
     var updated = false;
+    var removed = false;
 
-    var tagName = helpers.safeTagName('my-el');
-    skate(tagName.safe, {
-      attributes: {
-        open: {
-          created: function (element, data) {
-            created = true;
-            data.newValue.should.equal('created');
-            element.setAttribute('open', 'updated');
+    beforeEach(function () {
+      var MyEl = skate('my-el', {
+        attributes: {
+          camelCased: {
+            default: 'true'
           },
-          updated: function (element, data) {
-            updated = true;
-            data.oldValue.should.equal('created');
-            data.newValue.should.equal('updated');
-            element.removeAttribute('open');
+          'not-camel-cased': {
+            default: 'true'
           },
-          removed: function (element, data) {
-            created.should.equal(true);
-            updated.should.equal(true);
-            data.oldValue.should.equal('updated');
-            done();
+          'test-proxy': {},
+          'test-created': {
+            default: 'true',
+            created: function () {
+              created = true;
+            }
+          },
+          'test-lifecycle': {
+            created: function () {
+              created = true;
+            },
+            updated: function () {
+              updated = true;
+            },
+            removed: function () {
+              removed = true;
+            }
+          },
+          'override': {
+            default: 'false'
           }
+        },
+        prototype: {
+          override: 'true'
         }
-      }
+      });
+
+      myEl = new MyEl();
     });
 
-    helpers.fixture('<my-el open="created"></my-el>', tagName);
+    it('should respect attributes that are already camel-cased', function () {
+      myEl.camelCased.should.equal('true');
+    });
+
+    it('should camel-case the property name and leave the attribute name as is', function () {
+      myEl.notCamelCased.should.equal('true');
+    });
```

Comment:
> Should there be an assert to see if it leaves the attribute name as is?

Do you think that's necessary? There aren't any code paths that change the attribute name.